### PR TITLE
Display workstation version in `chef -v`

### PR DESCRIPTION
**Pull request by jonsmorrow**
### Description

This change monkey patches cli.rb from chefdk durning the omnibus build so that it shows chef workstation's version.

Signed-off-by: Jon Morrow <jmorrow@chef.io>
```
➜  omnibus git:(jm/patch_chef_cli) ✗ chef -v
Chef Workstation Version: 0.2.25+20181020052834
chef-dk version: 3.4.18
chef-apply version: 14.5.33
chef-client version: 14.5.33
delivery version: master (6862f27aba89109a9630f0b6c6798efec56b4efe)
berks version: 7.0.6
kitchen version: 1.23.2
inspec version: 2.2.112
```

### Check List

- [ ] PR title is a worthy inclusion in the CHANGELOG
- [ ] You have locally validated the change
- [ ] `www/site/content/docs/` has been updated with any relevant changes:
  - * new or changed error messages in 'troubleshooting.md'
  - * new or changed CLI flags in cli-reference.md

### CODE REVIEW

**Comment by jonsmorrow** [comment]
```
➜  omnibus git:(jm/patch_chef_cli) ✗ chef -v
Chef Workstation Version: 0.2.25+20181020052834
chef-dk version: 3.4.18
chef-apply version: 14.5.33
chef-client version: 14.5.33
delivery version: master (6862f27aba89109a9630f0b6c6798efec56b4efe)
berks version: 7.0.6
kitchen version: 1.23.2
inspec version: 2.2.112
```

File: omnibus/config/patches/chef-dk/cli.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
+diff --git a/lib/chef-dk/cli.rb b/lib/chef-dk/cli.rb
+index d789b075..adfc498e 100644
+--- a/lib/chef-dk/cli.rb
++++ b/lib/chef-dk/cli.rb
+@@ -95,9 +95,10 @@ BANNER
+     end
+ 
+     def show_version
+-      msg("Chef Development Kit Version: #{ChefDK::VERSION}")
++      msg("Chef Workstation Version: CHEF_WS_VERSION")
```
Comment (robbkidd) [inline omnibus/config/patches/chef-dk/cli.patch]: The former subversion user in me thinks that adding dollar signs here—`$CHEF_WS_VERSION$`—would make it more clear when looking at only this patch file that this is going to be replaced.

**Comment by cheeseplus** [comment]
One comment generally - maybe we remove the word "version" from every single line since it's super clear that is what `-v` or `--version` are asking for and it's a lot of redundant text

File: omnibus/config/patches/chef-dk/cli.patch [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ 
+-      ["chef-client", "delivery", "berks", "kitchen", "inspec"].each do |component|
++      msg("chef-dk version: #{ChefDK::VERSION}")
++      ["chef-apply", "chef-client", "delivery", "berks", "kitchen", "inspec"].each do |component|
```
Comment (marcparadise) [inline omnibus/config/patches/chef-dk/cli.patch]: In the screen shot, chef-apply seems to be the chef-client version, and not the chef-runply version. Namespace conflict w/ chef-apply in the client?

File: omnibus/config/patches/chef-dk/cli.patch [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
++++ b/lib/chef-dk/cli.rb
+@@ -95,9 +95,10 @@ BANNER
+     end
+ 
+     def show_version
+-      msg("Chef Development Kit Version: #{ChefDK::VERSION}")
++      msg("Chef Workstation Version: CHEF_WS_VERSION")
+ 
+-      ["chef-client", "delivery", "berks", "kitchen", "inspec"].each do |component|
++      msg("chef-dk version: #{ChefDK::VERSION}")
```
Comment (marcparadise) [inline omnibus/config/patches/chef-dk/cli.patch]: I don't think we should be showing ChefDK version, since this is a replacement for ChefDK.

**Comment by marcparadise** [comment]
+1 to naming it delivery-cli.

File: omnibus/config/patches/chef-dk/cli.patch (hunk 3 above)
Comment (cheeseplus) [reply omnibus/config/patches/chef-dk/cli.patch]: 👍

Comment (jonsmorrow) [reply omnibus/config/patches/chef-dk/cli.patch]: I can buy that. I guess my feeling was we are not yet replacing the chef cli experience which is why I didn't clean up the output either. Let me fix it up with no DK and I might also indent the components to show more visually they are part of ws.

File: omnibus/config/patches/chef-dk/cli.patch (hunk 2 above)
Comment (jonsmorrow) [reply omnibus/config/patches/chef-dk/cli.patch]: yup this is pulling the version from clients chef-apply... let me fix this to call chef-run.

**Comment by jonsmorrow** [comment]
```
Chef Workstation: 0.2.25+20181021212244.git.1.0a1398c
  chef-run: 0.2.0
  chef-client: 14.5.33
  delivery-cli: master (6862f27aba89109a9630f0b6c6798efec56b4efe)
  berks: 7.0.6
  test-kitchen: 1.23.2
  inspec: 2.2.112
```

**Pull request merged**